eduvpn/ui/app.py

```diff
@@ -129,6 +129,7 @@ def enter_ConnectedState(self, old_state, new_state):
         self.connection_notification.show(
             title=_("Connected"),
             message=_("You are now connected to your server."))
+        self.window.start_connection_info()
```

Comment:
The reason I added it here is that there doesn't seem to be a good way to check in ui.py if the connection is *actually* active. When I did it in ui.py in the connection switch I could get a bad state as we're starting the stats too early

Reply:
i might be wrong, but i guess the answer there is to have a callback mechanism that is triggered when the state is changed and then check if that state is ready. then you are sure you have a valid state. @alvra might already have state logic for that.

Reply:
You could define the same transition callback for on window, but it's better to call the start/stop in the enter/exit for a single transitions because its easier to check and maintain that it's correct. In your case you might need one more: enter/exit ConnectionStatus and when toggling the stats itself.

Reply:
Yeah that is the state transition here where I added it, I could add the same state transition to ui?

Reply:
> You could define the same transition callback for on window, but it's better to call the start/stop in the enter/exit for a single transitions because its easier to check and maintain that it's correct. In your case you might need one more: enter/exit ConnectionStatus and when toggling the stats itself.

whoops didn't see this, so should the stats toggling also be a separate state transition?